src/core/qgsinterval.h

```diff
@@ -187,6 +187,39 @@ class CORE_EXPORT QgsInterval
      */
     void setValid( bool valid ) { mValid = valid; }
 
+    /**
+     * Returns the original interval duration.
+     *
+     * The interval duration is set only through the
+     * QgsInterval constructors.
+     *
+     * This original interval duration doesn't change once set,
+     * it also does not reflect the current interval duration as
+     * QgsInterval duration can change through the available
+     * QgsInterval setter methods.
```

Comment:
Ok - a few more changes to make here:

1. Given that the "setMinutes", "setSeconds" etc methods all completely reset the interval to be a single time unit (i.e. calling setMonths with 3 months on an existing 5 day duration resets the interval to 3 months, it doesn't add to the existing 5 days to become 3months 5 days), there's no reason why we can't update the original duration and original units whenever these setters are called. That would allow us to remove this confusing behavior where the original duration may not have any resemblance at all to the interval size. (For bonus points, the various "setMinutes"/setHours setters should really explicitly mention that the existing interval size is replaced when the setter is called!) (also need tests for this)

2. The documentation for originalDuration() should mention something like "See originalUnit() for the corresponding unit. If the original duration is not available (e.g. the interval was set using a mix of units) then originalUnit() will return QgsUnitTypes::TemporalUnknownUnit."

3. The getters for months/years/days/ etc should return the original duration IF the original unit matches the requested unit. I.e. if I create a interval of 3 months duration, then calling months() on the interval should return 3.0, not the value calculated from the stored seconds value using the 30 day month assumption. (also need tests for this).  And again, the documentation for each getter should clearly state this. e.g. "Returns the interval duration in months. If the originalUnit() is QgsUnitTypes::TemporalMonths then this value will match the exact number of months as returned by originalDuration(), otherwise it will be calculated using the assumption that a month consists of exactly 30 days."